plaidbench/plaidbench/frontend_keras.py

```diff
@@ -68,17 +68,23 @@ def setup_imdb(train, epoch_size):
 
 class Model(core.Model):
 
-    def __init__(self, frontend, params):
+    def __init__(self, frontend, params, fold_bn_for_backend=False):
         learn_phase = params.learn_phase  # e.g. 0 for infer with fixed learning phase (but should come from param)
         if learn_phase is not None:
             from keras.backend import set_learning_phase
             set_learning_phase(learn_phase)
         self.frontend = frontend
         self.params = params
+        self.fold_bn_for_backend = fold_bn_for_backend
 
-    def fold_batch_norm(self, model):
+    def fold_batch_norm(self, model, backend):
         import json
-        import plaidml.keras.backend as K
```

Comment:
AFAICT, we only ever use `K.get_value()` and `K.set_value()`, so why can't we always just use `import keras.backend as K`? The assumption is that `import keras.backend` will route to the appropriate backend.

Reply:
`import keras.backend` does not actually route to the appropriate backend in the context of plaidbench in many circumstances due to technical debt lingering from when we had to monkey patch PlaidML into Keras. That said, it might be worthwhile to unwind that now rather than committing more hacky code like this

Reply:
At least in the context that this matters in (i.e. only the `plaid` and `plaid-edsl` backends), it appears to work properly (at least in the testing I did locally).

Reply:
hrm, it's been breaking my tests. I'm drilling into why though and it seems mostly unrelated, so I'm hoping I can clean this up.

Reply:
A major part of the broken code was adjusted in #527, so if you're not super up-to-date that might explain the discrepancy in our results